functions/mqhandlers.go

```diff
@@ -92,41 +85,23 @@ func NodeUpdate(client mqtt.Client, msg mqtt.Message) {
 	}
 	// Save new config
 	newNode.Action = models.NODE_NOOP
-	config.Nodes[network] = *newNode
+	config.UpdateNodeMap(network, *newNode)
 	if err := config.WriteNodeConfig(); err != nil {
 		logger.Log(0, newNode.Network, "error updating node configuration: ", err.Error())
 	}
-	nameserver := server.CoreDNSAddr
-	file := config.GetNetclientInterfacePath() + newNode.Interface + ".conf"
-
-	nc := wireguard.NewNCIface(newNode)
-	if newNode.ListenPort != newNode.LocalListenPort {
-		if err := nc.Close(); err != nil {
-			logger.Log(0, "error remove interface", newNode.Interface, err.Error())
-		}
-		err = config.ModPort(newNode)
-		if err != nil {
-			logger.Log(0, "network:", newNode.Network, "error modifying node port on", newNode.Name, "-", err.Error())
-			return
-		}
-		ifaceDelta = true
-		informPortChange(newNode)
-	}
-	if err := wireguard.UpdateWgInterface(file, nameserver, newNode); err != nil {
+	nc := wireguard.NewNCIface(config.Netclient(), config.GetNodes())
+	nc.Create()
```

Comment:
these should have error checks, maybe it isn't necessary right now, but in future

Reply:
added error handling